bundler.py

```diff
@@ -0,0 +1,194 @@
+            command=deps.cmd_with_node("yarn install"), 
+            cwd="account-abstraction",
+            log_file=log_file
+        )
+        # we need to configure the network to point to local network
+        # we also need to set private key for deployment
+        priv_key = input("Enter private key that you would like to deploy contracts with: ")
+        lib.run("set private key", f"echo PRIVATE_KEY={priv_key} > account-abstraction/.env")
+        log_file = "logs/deploy_4337_contracts.log"
+        lib.run_roll_log(
+            "deploy contracts", 
+            command=deps.cmd_with_node("yarn deploy --network opstack"), 
+            cwd="account-abstraction",
+            log_file=log_file
+        )
+        print("Account abstraction contracts successfully deployed.")
+    else:
+        print("Account abstraction contracts already deployed.")
+
+def setup_stackup_bundler():
+    github_url = "github.com/stackup-wallet/stackup-bundler"
+    version = "latest"
+
+    lib.run("installing stackup bundler", f"go install {github_url}@{version}")
+    print("Installation successful")
+
+    # set environment variables for bundler
+    lib.run("set full node RPC", "echo ERC4337_BUNDLER_ETH_CLIENT_URL=http://localhost:8545 > .env")
+    priv_key = input("Enter private key for bundler: ")
+    lib.run("set private key", f"echo ERC4337_BUNDLER_PRIVATE_KEY={priv_key} >> .env")
+
+    # make sure that GOPATH is set in PATH
+    current_path = os.environ.get("PATH", "")
+    gopath = subprocess.check_output(["go", "env", "GOPATH"]).decode().strip()
+    # append the bin directory of GOPATH to PATH
+    new_path = f"{gopath}/bin:{current_path}"
+    os.environ["PATH"] = new_path
+
+    # start bundler as a persistent process
+    print("Starting bundler...")
+    log_file_path = "logs/stackup_bundler.log"
+    log_file = open(log_file_path, "w")
+    PROCESS_MGR.start(
+        "start bundler", 
+        "stackup-bundler start --mode private",
+        forward="fd", 
+        stdout=log_file, 
+        stderr=subprocess.STDOUT
+    )
+    print("Bundler is running!")
+    # PROCESS_MGR.wait_all()
+
+def setup_paymaster():
+    # install paymaster dependencies
+    lib.run_roll_log(
+        "install paymaster dependencies", 
+        command=deps.cmd_with_node("npm install"), 
+        cwd="paymaster",
+        log_file="logs/install_paymaster_dependencies.log"
+    )
+
+    # set environment variables for paymaster (deterministic deployments can be hardcoded)
+    lib.run("set node RPC", "echo RPC_URL=http://localhost:8545 > paymaster/.env")
+    lib.run("set paymaster RPC", "echo PAYMASTER_RPC_URL=http://localhost:3000 >> paymaster/.env")
+    lib.run(
+        "set entrypoint", 
+        "echo ENTRYPOINT_ADDRESS=0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789 >> paymaster/.env"
+    )
+    lib.run(
+        "set factory", 
+        "echo SIMPLE_ACCOUNT_FACTORY_ADDRESS=0x9406Cc6185a346906296840746125a0E44976454 >> paymaster/.env"
+    )
```

Comment:
Are these always fixed or must we save them during deployment and reimport them here?

Even if there's CREATE3 to do deterministic deployment, the contract might change when stackup updates, so probably some automation here would be good.

Reply:
These addresses will be constant even with stackup updates. They will only change if eth foundation decides to release a new version of the contracts, which in that case we will need to make some changes in the `bundler.py` script anyways (because we clone from a specific version of the account abstraction contracts).

But yeah I guess we could still read the address directly from the artifacts.

Reply:
Ok, have now automated these addresses